File: Common/src/main/java/com/example/examplemod/CommonClass.java
package com.example.examplemod;

import com.example.examplemod.platform.Services;
import net.minecraft.core.Registry;
import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.TextComponent;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.food.FoodProperties;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import net.minecraft.world.item.TooltipFlag;

import javax.annotation.Nullable;
import java.util.List;

public class CommonClass {

    // This method serves as an initialization hook for the mod. The vanilla
    // game has no mechanism to load tooltip listeners so this must be
    // invoked from a mod loader specific project like Forge or Fabric.
    public static void init() {

        Constants.LOG.info("Hello from Common init on {}! we are currently in a {} environment!", Services.PLATFORM.getPlatformName(), Services.PLATFORM.isDevelopmentEnvironment() ? "development" : "production");
        Constants.LOG.info("Diamond Item >> {}", Registry.ITEM.getKey(Items.DIAMOND));
    }

    // This method serves as a hook to modify item tooltips. The vanilla game
    // has no mechanism to load tooltip listeners so this must be registered
    // by a mod loader like Forge or Fabric.
    public static void onItemTooltip(ItemStack stack, @Nullable Player player, TooltipFlag context, List<Component> tooltip) {

        if (!stack.isEmpty()) {

            final FoodProperties food = stack.getItem().getFoodProperties();

            if (food != null) {

                tooltip.add(new TextComponent("Nutrition: " + food.getNutrition()));
                tooltip.add(new TextComponent("Saturation: " + food.getSaturationModifier()));
            }
        }
    }
}

File: MultiLoader-Template-with-kotlin-and-kotlin-dsl/Common/src/main/kotlin/com/example/examplemod/CommonClass.kt
package com.example.examplemod

import com.example.examplemod.platform.Services
import net.minecraft.core.Registry
import net.minecraft.network.chat.Component
import net.minecraft.network.chat.TextComponent
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import net.minecraft.world.item.TooltipFlag

class CommonClass {
    companion object {
        // This method serves as an initialization hook for the mod. The vanilla
        // game has no mechanism to load tooltip listeners so this must be
        // invoked from a mod loader specific project like Forge or Fabric.
        fun init() {
            Constants.LOG.info(
                "Hello from Common init on {}! we are currently in a {} environment!",
                Services.PLATFORM.getPlatformName(),
                if (Services.PLATFORM.isDevelopmentEnvironment()) "development" else "production"
            )
            Constants.LOG.info("Diamond Item >> {}", Registry.ITEM.getKey(Items.DIAMOND))
        }

        // This method serves as a hook to modify item tooltips. The vanilla game
        // has no mechanism to load tooltip listeners so this must be registered
        // by a mod loader like Forge or Fabric.
        fun onItemTooltip(stack: ItemStack, context: TooltipFlag?, tooltip: MutableList<Component?>) {
            if (!stack.isEmpty) {
                val food = stack.item.foodProperties
                if (food != null) {
                    tooltip.add(TextComponent("Nutrition: " + food.nutrition))
                    tooltip.add(TextComponent("Saturation: " + food.saturationModifier))
                }
            }
        }
    }
}